core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -443,6 +439,10 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime) extends Logg
         case _ => // we just leave those out
       }
     }
+    logProps.put(LogConfig.SegmentMsProp, kafkaConfig.logRollTimeMillis)
```

Comment:
I think its now a bit messy and unclear why some parameters are set in the switch statement and some outside.

Either comment on the logic here or unify these?